lib/PrintAsClang/PrintClangFunction.cpp

```diff
@@ -157,10 +166,16 @@ void DeclAndTypeClangFunctionPrinter::printFunctionSignature(
     // see DeclAndTypePrinter::print.
     CFunctionSignatureTypePrinter typePrinter(os, cPrologueOS, typeMapping,
                                               outputLang, interopContext);
-    typePrinter.visit(ty, optionalKind);
+    typePrinter.visit(ty, optionalKind, isInOutParam);
 
     if (isInOutParam) {
-      os << (outputLang == OutputLanguageMode::Cxx ? " &" : " *");
+      if (outputLang == OutputLanguageMode::Cxx &&
+          isKnownCxxType(ty, typeMapping)) {
```

Comment:
Suggestion: can you move the `&` / `*` printer for primitive types into the visitor as well, to be more consistent with the inout types for struct types?  You can add the `bool` flag to `printIfKnownSimpleType` and handle it in there.